### PR TITLE
Move ReadonlyFilesystem Node Condition to a new configuration file

### PR DESCRIPTION
**Pull request by DigitalVeer**
We're enhancing NPD to support detection of various read-only scenarios (e.g., boot disk, local SSDs, network-attached drives). To support this, the ReadonlyFilesystem configuration is being moved from the kernel monitor into a dedicated plugin configuration file that will take over this new functionality.

### CODE REVIEW

**Comment by mmiranda96** [comment]
You need to add your new plugin config file to https://github.com/kubernetes/node-problem-detector/blob/dc4200d805c9554d6585dba8d1a566fb20b6869b/config/systemd/node-problem-detector-metric-only.service#L11 (and potentially other files).

**Comment by wangzhen127** [comment]
To make sure it is backward compatible for people using the OSS config file directly, we need to add the new file to all the locations where kernel-monitor.json is invoked: https://github.com/search?q=repo%3Akubernetes%2Fnode-problem-detector%20kernel-monitor.json&type=code

**Review by mmiranda96 — APPROVED** [review]
/lgtm

Thanks for your contribution!

File: config/readonly-monitor.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+	"plugin": "kmsg",
+	"logPath": "/dev/kmsg",
+	"lookback": "5m",
+	"bufferSize": 10,
+	"source": "kernel-monitor",
```
Comment (wangzhen127) [inline config/readonly-monitor.json:6]: readonly-monitor

**Comment by hakman** [comment]
@DigitalVeer please rebase and squash the commits.

**Comment by hakman** [comment]
@wangzhen127 There's a problem with the failing tests. Seems to be something broken in some other related repo. Any ideas?

**Pull request merged**